### DISCUSSION TITLE
Does an agent in a process need to be causally active within that process?

### DISCUSSION THREAD

**Discussion by nklsbckmnn** [CommonCoreOntology/CommonCoreOntologies#397] [Announcements]
has agent is currently defined as follows: "As for has_participant, but with the additional condition that the component instance is causally active in the relevant process." 

Is it sufficient to causally influence a process upstream (temporally) to be „causally active in“ it or is it neccessary to be causally efficious within it?

See http://www.ontobee.org/ontology/RO?iri=http://purl.obolibrary.org/obo/RO_0002500 for the distincitions that have been made in RO.

**Comment by rorudn** [comment]
As a sub-property of the BFO has_participant and participates_in properties, use of the has_agent and agent_in properties implies that the agent is "...somehow involved in the process." That leaves some amount of latitude for interpretation but I use the has_agent and agent_in properties for relations between an agent and occurrent to which the agent is causally efficacious. I then link the downstream causal consequences of the occurrent using the is_cause_of and caused_by properties.

**Comment by nklsbckmnn** [comment]
I use it the same way.

But it still would be great if the definition made unambiguous whether this latitude for interpretation is supposed to exist or not.

E. g. it might be useful to change it to sth. like

"As for has_participant, but with the additional condition that the component instance is causally active within the relevant process.“

or something like

"As for has_participant, but with the additional condition that the component instance is causally active within the relevant process or within another process causally upstream of that process.“

**Comment by rorudn** [comment]
Fair point. We'll improve the definition in the next release.

**Comment by alanruttenberg** [comment]
How is this different than any participant? "Causally active" is hard to evaluate because "cause" is difficult to define unambiguously.  If there is to be an agent term, which I generally discourage because of these sorts of problems, I would have the business end be in an agent *role*. Definitions of subclasses of agent role, which elaborate on bearer, realizing process, and manner of participation, can help clarify exactly what an agent role is.  The Agent class can then be defined as bearing the Agent role, and x agent in y = x has an agent role that is realized in y.

I know this is an old issue. I'm adding a comment now triggered by @eliasweatherfield's comment on #137.

IMO, definitions in terms of intention are more easy to evaluate. The source of truth for whether there is intention is the participant, and at least you can _ask_ the participant about it, unless the participant is an animal. Overall still easier than determining whether a participant is "causally active".

**Comment by nklsbckmnn** [comment]
> How is this different than any participant?

There are a lot of subrelations of participates in CCO that don't require the participant to be causally active within the process. "is affected by" is one example.

> IMO, definitions in terms of intention are more easy to evaluate.

I agree.

Dropping the (implicit) "causally active in" relation would mean that we have to have a lot of roles like "agent role" because it is not only agents that can be causally active within a process. We also lose the ability to distinguish between causal influence upstream of and within a process unless we disambiguate this in the definition of the role. I don't really see the problems with ambiguity that you speak of. Can't different causal theories get their own subrelation so it's clear what causal theory is used? Do you have a specific example?

**Comment by alanruttenberg** [comment]
The following process of making supper: <start>I light a fire and cook a steak heat goes from the fire to the steak. The steak chars and I cough from the smoke. My wife grabs a plate puts the steak in it, where it cools down, but she singes herself by mistake as she reaches around me, and it hurts. Cranky because of that, she yells at me for not bringing me the right utensils, which hurts my feelings. Heat goes from the steak to the plate, cooling the steak, warming the plate and leaking juice that will have to be washed off.<stop>

Which things are causally active? Is there anything that isn't causally active?  Which things are agents? How much, in making that judgement, are you relying on the way I've _described_ the process as compared just having only _seen_ what happens. What is the reasoning behind your assessment?

Are you suggesting that there may be different causal theories? Is anything causally active according to _any_ of the theories sufficient to assert that an entity is causally active in the process? Suppose that I'm religious and assert there is no agency at all, only god's will that determines what happens? 

Compare to the intentional view. In that view there are goals which someone wants to achieve and they undertake actions in order to achieve the goals. During the undertaking there are processes that are deliberately engaged because the intention bearer wants to achieve the goal and considers a process a means to towards accomplishment, are intentional processes. The intention resides in the thing that accepts the goal and plans or acts to achieve it. An analysis in these terms rules out many of the participants in the process/es I've described above.

Don't get me wrong - there are still problems to address, such as where the boundaries between processes are. But such questions are relevant independent of the causal analysis so overall we've made some headway.

If there is a specific causal theory with which we can make unambiguous judgements of causal activity, then I would define that separately. It's not clear to me that there would be anything in common among those theories. So define relations/roles for specific cases and only after those exist and it becomes apparent that there is some commonality should a generalization be considered.

The closest thing to cause in BFO is the relationship between an entity bearing a realizable and the process in which that is realized. Since that's already there in BFO my strategy would be to leverage it, which is where my suggestions originate.

If you want to distinguish upstream vs current process then a reasonable way to represent it might be to talk about a) The bearer/realizable/realization in each process and b) A realizable that is created in earlier processes (or a pre-existing realizable whose conditions for being realized come to be)  and realized in subsequent ones. Remember that many realizables may be realized in a single process. Then, looking at the process p you are focusing on, the bearer of the realizable realized in p is immediate, and any participant in a chain backwards in time starting at p and following the chain of realization creation/enabing in a realization gives you upstream.

**Comment by nklsbckmnn** [comment]
> Which things are causally active? Is there anything that isn't causally active? Which things are agents? How much, in making that judgement, are you relying on the way I've described the process as compared just having only seen what happens. What is the reasoning behind your assessment?

If this is represented as just one single process (which is not very useful), then you, your wife, the fire, the steak, the plate are all causally active within the process. Not all of these instances are causally active in all of the parts of the process though.

> Are you suggesting that there may be different causal theories? Is anything causally active according to any of the theories sufficient to assert that an entity is causally active in the process? Suppose that I'm religious and assert there is no agency at all, only god's will that determines what happens?

Yes. There are normative theories of causality where the definition of causally active within is narrower (requiring blameworthiness) than natural causality (sine qua non). Since natural causality is implied, there is a commonality.

> Compare to the intentional view. In that view there are goals which someone wants to achieve and they undertake actions in order to achieve the goals. During the undertaking there are processes that are deliberately engaged because the intention bearer wants to achieve the goal and considers a process a means to towards accomplishment, are intentional processes. The intention resides in the thing that accepts the goal and plans or acts to achieve it. An analysis in these terms rules out many of the participants in the process/es I've described above.

I think we need the intentional view, too. We can have a subrelation of "causally active in" for causal activity that is intentional and define that by Directive ICEs.

> If you want to distinguish upstream vs current process then a reasonable way to represent it might be to talk about a) The bearer/realizable/realization in each process and b) A realizable that is created in earlier processes (or a pre-existing realizable whose conditions for being realized come to be) and realized in subsequent ones. Remember that many realizables may be realized in a single process. Then, looking at the process p you are focusing on, the bearer of the realizable realized in p is immediate, and any participant in a chain backwards in time starting at p and following the chain of realization creation/enabing in a realization gives you upstream.

Yes, we could do that. What I wanted to point out is that I think it is not clear from the definition of realizes that causation upstream is excluded. And roles can also be passively (non-causally) realized. So in the definition of the roles we might need to talk about being "causally active" again. And then we have just moved the problem.

**Comment by nklsbckmnn** [comment]
Another problem is that there are inferentialist and subjective accounts of causality that say that it can't always be defined objectively and is e. g. dependent on domain and interest. Causality according to different accounts might not have commonalities, but causality between domains/interests according to one specific account does have commonalities. I don't see the benefit of having different roles for different kinds of causality over having different relations.

**Comment by nklsbckmnn** [comment]
Take this inferentialist account by Julian Reiss: "Causation is fundamentally subjective in that what warrants the appearance of the term ‘cause’ or a cognate in a claim is the claim’s being inferentially related in certain ways to certain other kinds of claim, and to be inferentially related is to play a role in—human—reasoning practices. But, and this is what makes causation partly objective, these reasoning practices may not be arbitrary but rather shaped by objective facts. [...] [T]he objectivity of causal claims comes in degrees, and some sciences may establish causal claims of higher, others of lesser objectivity." The objective facts mentioned are standards of explanation and prediction in a domain.

**Comment by nklsbckmnn** [comment]
So I think my point is this: Your proposal seems to move the ambiguity in the relation (that I now agree exists) to the role.

By the way, I think we have similar problems with "causes" and "caused by".

**Comment by alanruttenberg** [comment]
If the only role is "agent role" then I agree ambiguity has just been moved without solving anything. Even in that case, though, it is still a good move. Participation itself is not well defined and adding a fuzzy distinction to an already fuzzy term just compounds the problem. I'm not suggesting that you use the general agent role directly. Rather, it's a way to minimize damage in the case that someone is drawn to use the term. So my suggestion is to create better-defined subclasses of agent role.  That's why I recommend defining more specific roles for which there's a reasonable chance of conveying a clear understanding of the term. If one must use _agent_, then if it is considered a union of the more specific terms then you at least have enumerated the possibilities.  If agent role is a defined class, the heavy definitional lifting is done by the subclasses.

Two other considerations play in the suggestion: In OWL, and to some extent in FOL, there is greater expressivity when defining in terms of classes than relations, so there's an engineering reason to make the choice. From an ontological perspective we have a precedent for "explaining" why and how something participates in a process. As in law, precedent should be respected.

The consequence of there not being a clear/useful definition of agent role is that if you are integrating data from different sources the different sources will likely use different criteria for deciding there's an "agent" involved. So, the results of any query that depends on agent will not reliably the querier what they want. It's hard enough to do representation consistently when there *are* good definitions. But the more choice there is in whether to use a term or not, the worse the consistency will be. Agent is one of the more ambiguous terms I run into. I have had discussions about what an agent is over the years and I can tell you. at least anecdotally, that people have different ideas about what an agent is. Sometimes radically. *agent* and *cause* are dangerous because everybody thinks they know what the term means but don't have a lot of perspective on what it might mean to others. If I represent a process as an engineer, will the same judgements be made about agent/causes by a sociologist? By a chemist? My experience says "sometimes" but often not.

If one is doing representation for a specific task without any intention of sharing or of someone later integrating your representations, then one can do what they want. I'm just not interested in that case and assume, in everything I write here, that we're discussing building representations that *are* to be shared and integrated.

I usually approach processes and participation with the null hypothesis that all participants have things happen to them and so are on an equal footing in that sense. I also try to take the perspective of different people who might be observing the process. If I represent a process as an engineer, will the same judgements be made about agent/causes by a sociologist? By a chemist? My experience says "sometimes" but often not. Consider seeing an explosion. What is the agent. If you are a rescue worker you might consider what injures people as an agent. If you are political you might consider whoever physically triggered the explosion as agent. If you are a political scientist you might consider the group which decides to send the person as the agent. In case you think this can be solved by distinguishing present versus upstream cause, what's the story with a IED? Who or what is an agent in that case? The situation that I try to anticipate is that all of these cases might be representations of the same process. Since all of them choose "agent" to be a plausible, how much use is a query for the "agent" going to be. So, I try to see whether there is a way to distinguish the participation in a way that all viewers have the possibility of making the same judgement of what happened. 

The inferentialist account you describe is reasonably close to how I view things. To the extent that objective facts underlie a judgement of cause and pre-exist the process, I'd suggest representing roles, dispositions, functions that capture those facts.  Have a superclass agent role if you must, but my read of it will be: a participant that some author of some ontology has singled out for some reason. That's not going to be useful for practical purposes unless the interest is in trying to tease out the difference between authors. But explicit representations of who makes a judgement and why has a fighting chance at being understood in the same way by different users.

There are several approaches consider practical. In OBI, we have has_specified_input and has_specified_output because input and output simpliciter are hard to judge objectively, but whether or not someone has made a choice to designate some participants is. That's what "specified" captures. It's still not great but it's better because it explicitly acknowledges the role of choice in the designation.

Another way is to think in terms of time. Has input sometimes is defined as a process existing at the beginning of a process, but I don't think that captures the idea well. A first pass approach to deciding participation is to ask if something is happening to an entity that is part of what you are calling a process. So consider cooking from a recipe. I might argue that the ingredients don't participate in my cooking until I start preparing them or adding them to what I'm cooking, and that doesn't mean they are participating from the start. At the same time, I think the ingredients are what would commonly be designated as input. But I can reliably say that if I'm making phyllo dough, then before I can roll out the dough, I have to have gathered the flour, and only after some rolling can the folding happen.

Some things that might be said: some thing starts participating *after* some other thing participates. Or, something might start existing only after a process starts, for example when there is a chemical reactions. Or you might say that something necessarily exists at the end of a process. Or you might say that every process of type A with participant p always follows another process of type B with the participant p. Each of these temporal statements has a better chance of being objectively determined and verified than whether something is (in a general sense) active or passive. If there's a process where something is build from parts, then the presence of at least some of the parts precedes the building, and that the building isn't complete until all the parts have been present. If you want to capture that the person who is doing the building is "agent", better to represent the role of "builder" and then document what that means.

Maybe there's a way to consistently and reliably talk about causality. If it ever becomes pressing I'd go read Judea Pearls book "Causality", and brush up on Bayesian statistics as, from what I know, that's where the useful stuff is.

Summary: There are a variety of ways that one might capture ideas of activity activity causality that are much less problematic and much easier to be objective about. Why not use them?

 [btw, in most places I write about roles, it's as an exemplar of realizable entity not to distinguish between roles and dispositions]

**Comment by neilotte** [comment]
There is a related duplicate post to the original request to this issue, which is captured in https://github.com/CommonCoreOntology/CommonCoreOntologies/issues/186 and will be addressed in an upcoming release.

Many of the other items here warrant further discussion. For this reason, converting this issue now and will address the original request under 186.
